sympy/sets/fancysets.py

```diff
@@ -472,14 +469,21 @@ def __new__(cls, *args):
         if start.is_infinite:
             end = stop
         else:
-            ref = start if start.is_finite else stop
-            n = ceiling((stop - ref)/step)
-            if n <= 0:
-                # null Range
-                start = end = S.Zero
-                step = S.One
+            if not all(w.is_number for w in [start, stop, step]):
```

Comment:
```
if not all(i.is_Symbol or i.is_Integer or i.is_infinite for i in (ref, stop, step)):
    raise error
```

Reply:
I think you misplaced the check. Shouldn't it be in else part?

Reply:
Ah! I got it.

Reply:
I am supportive of using `atoms` as `is_attr` may require fuzzy logic.

Reply:
check to code again...previous lines are handling the infinities. And sanitize the input -- var('inf', infinite=True) should be rejected unless the sign is known; when the sign is know it should be replaced with a literal oo or -oo.